server/main_server/src/rodsServer.cpp

```diff
@@ -271,6 +271,68 @@ namespace
         }
     }
 
+    void remove_leftover_shared_memory_files() noexcept
+    {
+        using log_server = irods::experimental::log::server;
+
+        try {
+            namespace fs = boost::filesystem;
+
+            std::vector<fs::path> paths_to_remove;
+            paths_to_remove.reserve(10);
+
+            // clang-format off
+            const auto shm_dirs = {
+                fs::path{"/dev/shm"},
+                fs::path{"/var/run/shm"},
+                fs::path{"/run/shm"}
+            };
+            // clang-format on
+
+            for (auto&& shm_dir : shm_dirs) {
+                try {
+                    if (!fs::exists(shm_dir)) {
+                        log_server::trace("[{}] directory does not exist. Skipping.", shm_dir.c_str());
+                        continue;
+                    }
+
+                    paths_to_remove.clear();
+
+                    // Build up a list of files to remove.
+                    // This avoids potential issues with the directory iterator.
+                    for (auto&& shm_file : fs::directory_iterator{shm_dir}) {
+                        const auto filename = shm_file.path().filename();
+                        const std::string_view s = filename.c_str();
+                        const std::string_view t = "irods";
+                        const auto e = std::end(s);
+
+                        if (std::search(std::begin(s), e, std::begin(t), std::end(t)) != e) {
```

Comment:
does this assert that it begins with "irods_"... or just that it is in there somewhere?

i think we want to assert 'begins with'.